packages/@angular/cli/plugins/named-lazy-chunks-webpack-plugin.ts

```diff
@@ -0,0 +1,38 @@
+import * as webpack from 'webpack';
+
+// This is just a wrapper around webpack.NamedChunksPlugin to prevent name collisions.
+export class NamedLazyChunksWebpackPlugin {
+  constructor() {
+    const nameMap = new Map();
+    // Append a dot and number if the name already exists.
+    const getNextName = (baseName: string) => {
+      let num: number | null = null;
+      const name = () => num !== null ? `${baseName}.${num}` : baseName;
+      while (nameMap.has(name())) {
+        num = num ? num++ : 0;
+      }
+      nameMap.set(name(), true);
+      return name();
+    };
+
+    return new (webpack as any).NamedChunksPlugin(function (chunk: any) {
```

Comment:
I don't think it's best practice for a constructor to return another class. Could you instead `extends` `NamedChunksPlugin`?